src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -208,18 +208,20 @@ internal IOperation CreateMemberInitializerInitializedMember(BoundNode initializ
             }
         }
 
+#nullable enable
         internal ImmutableArray<IArgumentOperation> DeriveArguments(BoundNode containingExpression, bool isObjectOrCollectionInitializer)
         {
             switch (containingExpression.Kind)
             {
                 case BoundKind.ObjectInitializerMember:
                     {
                         var boundObjectInitializerMember = (BoundObjectInitializerMember)containingExpression;
-                        var property = (PropertySymbol)boundObjectInitializerMember.MemberSymbol;
-                        MethodSymbol accessor = isObjectOrCollectionInitializer ? property.GetOwnOrInheritedGetMethod() : property.GetOwnOrInheritedSetMethod();
+                        var property = (PropertySymbol?)boundObjectInitializerMember.MemberSymbol;
+                        MethodSymbol? accessor = isObjectOrCollectionInitializer ? property.GetOwnOrInheritedGetMethod() : property.GetOwnOrInheritedSetMethod();
+                        Debug.Assert(property is not null);
```

Comment:
>Debug.Assert(property is not null); [](start = 24, length = 35)

Should the assert be moved right after the declaration of ```property```? It looks like it is dereferenced in the previous statement.

Reply:
No. Those are extension methods that accept a `null` input. I could certainly move it, but it doesn't particularly matter.

Reply:
> Those are extension methods that accept a null input. I could certainly move it, but it doesn't particularly matter.

I think that would benefit the reader.

---
In reply to: [497941233](https://github.com/dotnet/roslyn/pull/48168#discussion_r497941233) [](ancestors = 497941233)